core/common/src/main/java/alluxio/exception/AlluxioRuntimeException.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.exception;
+
+import alluxio.grpc.RetryInfo;
+
+import com.google.common.base.Preconditions;
+import com.google.protobuf.Any;
+import io.grpc.Metadata;
+import io.grpc.Status;
+import io.grpc.StatusException;
+import io.grpc.protobuf.ProtoUtils;
+
+/**
+ * Alluxio RuntimeException. Every developer should throw this exception when need to surface
+ * exception to client.
+ */
+public class AlluxioRuntimeException extends RuntimeException {
```

Comment:
Ported over when cherry-picking https://github.com/Alluxio/alluxio/pull/15741

Didn't want to include https://github.com/Alluxio/alluxio/pull/15611 if it can be avoided.